### PR TITLE
Merge n-way symmetric migrations

### PR DESCRIPTION
**Pull request by apragsdale**
I don't know if I've made this overly complicated, but the algorithm here in the `simplify_migration_rates()` function in `asdict_simplified()` iteratively searches for sets of n-way symmetric migrations within a given migration rate attribute set (rate, start_time, end_time), in a way that hopefully catches all n-way symmetric migrations, starting with those that include larger sets of demes.

It needs some more testing, but there are a few simple test scenarios in `test_demes.py`.

### CODE REVIEW

File: demes/demes.py [hunk 1]
```diff
@@ -1224,20 +1242,68 @@ def simplify_migration_rates(data):
...
-                    sym = dict(demes=[dest, source], **opposite)
-                    del sym["source"]
-                    del sym["dest"]
-                    symmetric.append(sym)
+
+                k = [migration["rate"]]
+                try:
+                    k.append(migration["start_time"])
+                except KeyError:
+                    k.append(None)
```
Comment (grahamgower) [inline demes/demes.py:1250]: Here, you can use the idiom `k.append(migration.get("start_time"))` instead.

Comment (grahamgower) [reply demes/demes.py:1250]: Or indeed `k = tuple(migration.get(key) for key in ("rate", "start_time", "end_time"))`

**Review by grahamgower — APPROVED** [review]
It's a bit complicated, but much less so than I was expecting. Nice job!

File: demes/demes.py (hunk 1 above)
Comment (apragsdale) [reply demes/demes.py:1250]: Oh, nice! Didn't know I could do that. Thanks

**Pull request merged**